functions/Get-DbaDbccMemoryStatus.ps1

```diff
@@ -0,0 +1,106 @@
+
+    .NOTES
+        Tags: DBCC, Memory
+        Author: Patrick Flynn (@sqllensman)
+
+        Website: https://dbatools.io
+        Copyright: (c) 2018 by dbatools, licensed under MIT
+        License: MIT https://opensource.org/licenses/MIT
+
+    .LINK
+        https://dbatools.io/Get-DbaMemoryStatus
+
+    .EXAMPLE
+        PS C:\> Get-DbaDbccMemoryStatus -SqlInstance sqlcluster, sqlserver2012
+
+        Get output of DBCC MEMORYSTATUS for instances "sqlcluster" and "sqlserver2012". Returns results in a single recordset.
+
+    .EXAMPLE
+        PS C:\> Get-DbaCmsRegServer -SqlInstance sqlcluster | Get-DbaDbccMemoryStatus
+
+        Get output of DBCC MEMORYSTATUS for all servers in Server Central Management Server
+
+          #>
+    [CmdletBinding()]
+    param (
+        [parameter(Mandatory, ValueFromPipeline)]
+        [DbaInstanceParameter[]]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [switch]$EnableException
+    )
+    begin {
+        $query = 'DBCC MEMORYSTATUS'
+    }
+    process {
+        foreach ($instance in $SqlInstance) {
+            Write-Message -Message "Attempting Connection to $instance" -Level Verbose
+            try {
+                $server = Connect-SqlInstance -SqlInstance $instance -SqlCredential $SqlCredential
+            } catch {
+                Stop-Function -Message "Failure" -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
+            }
+
+            Write-Message -Level Verbose -Message "Collecting $query data from server: $instance"
+            try {
+                $sqlconnection = New-Object System.Data.SqlClient.SqlConnection
+                $sqlconnection.ConnectionString = $server.ConnectionContext.ConnectionString
+                $sqlconnection.Open()
+
+                $datatable = New-Object system.Data.DataSet
+                $dataadapter = New-Object system.Data.SqlClient.SqlDataAdapter($query, $sqlconnection)
+                $dataadapter.fill($datatable) | Out-Null
+
+                $recordset = 0
+                $rowId = 0
+                $recordsetId = 0
+
+                foreach ($dataset in $datatable.Tables) {
+                    $dataSection = $dataset.Columns[0].ColumnName
+                    $dataType = $dataset.Columns[1].ColumnName
+                    $recordset = $recordset + 1
+                    foreach ($row in $dataset.Rows) {
+                        $rowId = $rowId + 1
+                        $recordsetId = $recordsetId + 1
+                        [PSCustomObject]@{
+                            ComputerName = $server.ComputerName
+                            InstanceName = $server.ServiceName
+                            SqlInstance  = $server.DomainInstanceName
+                            RecordSet    = $RecordSet
+                            RowId        = $RowId
+                            RecordSetId  = $RecordSetId
+                            MemObjType   = $dataSection
```

Comment:
Can these be `Type, Name, Value, ValueType`? If not, then please write out everything `MemoryObjectName` etc

Reply:
Type, name value are fine. The names were taken from original article.

Reply:
Type, name value are fine. The names were taken from original article.

Reply:
Type, name value are fine. The names were taken from original article.